### DISCUSSION TITLE
Struggling to get sufficient permissions to run the webpart

### DISCUSSION THREAD

**Discussion by MMAtHome** [YannickRe/spfx-msgraph-peoplesearch#57] [Q&A] [answered]
Hi, our IT team have implemented MFA and I am struggling to grant the required permissions for the webpart.  Having connected to PnP and attempting to run the relevant commands, I get the following error.

We are using an account that is global admin.

PS C:\Windows\system32> Grant-PnPTenantServicePrincipalPermission -Scope "User.Read.All" -Resource "Microsoft Graph"

Grant-PnPTenantServicePrincipalPermission : Access is denied. (Exception from HRESULT: 0x80070005 (E_ACCESSDENIED))
At line:1 char:1
+ Grant-PnPTenantServicePrincipalPermission -Scope "User.Read.All" -Res ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Grant-PnPTenant...cipalPermission], ServerUnauthorizedAccessException
    + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.Apps.GrantTenantServicePrincipalPermission

**Comment by YannickRe** [comment]
It looks like your account is not having the correct permissions, which would be impossible if it is a Global Admin...
Could you try the M365 CLI and see if you have the same issue?

**Comment by MMAtHome** [comment, marked as answer]
Yannick,

Once again, thank you very much for your help on this issue.

The issue appears to have been down to the permissions that the "admin" had
available.  He did not have global tenant privileges hence the request was
coming back as "unauthorised".  I realise this is out-of-scope of what you
write on Github, but I wonder if it may be useful to those who may be new
to AzureAD / GraphAPI.

Thanks again,

Martin

**Reply by YannickRe** [reply]
Good you got it resolved, and thank you for getting back on the solution.
